src/Hal/Serializer/ItemNormalizer.php

```diff
@@ -205,16 +223,31 @@ private function populateRelation(array $data, object $object, ?string $format,
                 continue;
             }
 
-            $attributeValue = $this->getAttributeValue($object, $relation['name'], $format, $context);
-            if (empty($attributeValue)) {
-                continue;
-            }
-
             $relationName = $relation['name'];
             if ($this->nameConverter) {
                 $relationName = $this->nameConverter->normalize($relationName, $class, $format, $context);
             }
 
+            // if we specify the uriTemplate, then the link takes the uriTemplate defined.
+            if ('links' === $type && $iri = $relation['iri']) {
+                $data[$key][$relationName]['href'] = $iri;
+                continue;
+            }
+
+            $childContext = $this->createChildContext($context, $relationName, $format);
```

Comment:
What exactly is the purpose of this and the next few lines? `$relation['operation']` is only populated when `$relation['iri']` is populated as well, in which case the for-loop has already continued/returned above.

Reply:
I'll double check that, I may have been carried away x)

Reply:
I think for HAL and JSON:API formats, the behaviour is fine.

But for the JSONLD format, I feel that we should not use the uriTemplate acting as an IriOnly option, when there is one that exist for NormalizationContext. Would it make more sense that, **unless we use a second `ApiProperty::IriOnly` option**, it only resolves the IRI with uriTemplate within the object(s) themselves ? And if the `ApiProperty::IriOnly` option is true, **then** return the IRI string instead of an object? But then what about the IriOnly option on Normalization Context?

ping @vincentchalamon @dunglas

Reply:
As discussed on video call with Vincent and Soyuka, the behaviour is different, but doesn't interfere that much. It is acceptable.